tests/e2e/etcd_release_upgrade_test.go

```diff
@@ -105,12 +104,18 @@ func TestReleaseUpgrade(t *testing.T) {
 
 	// TODO: update after release candidate
 	// expect upgraded cluster version
-	ver := version.Version
-	if strings.HasSuffix(ver, "-pre") {
-		ver = strings.Replace(ver, "-pre", "", 1)
+	// new cluster version needs more time to upgrade
+	ver := version.Cluster(version.Version)
+	for i := 0; i < 7; i++ {
+		if err = cURLGet(epc, cURLReq{endpoint: "/version", expected: `"etcdcluster":"` + ver}); err != nil {
+			t.Logf("#%d: %v is not ready yet (%v)", i, ver, err)
+			time.Sleep(time.Second)
+			continue
+		}
+		break
 	}
-	if err := cURLGet(cx.epc, cURLReq{endpoint: "/metrics", expected: fmt.Sprintf(`etcd_cluster_version{cluster_version="%s"} 1`, ver), metricsURLScheme: cx.cfg.metricsURLScheme}); err != nil {
```

Comment:
Prior to your change, the test was using the /version endpoint to check the version before upgrade then use the /metrics endpoint to check the version afterward. I think it's better with your change that we check the version with the consistent way. But I am just curious, @gyuho is there specific reason that we previous check the /metrics endpoint afterward? Just want to make sure we are not missing testing anything.

Reply:
@wenjiaswe I think either `/version` or `/metrics` is fine, since they are all set by the same constant (`version.Version`). Since this is just upgrade test, using `/version` makes more sense.